File: android/app/src/main/java/vn/team/freechat/data/MessageReceived.java
package vn.team.freechat.data;

import com.tvd12.ezyfoxserver.client.constant.EzyConstant;
import com.tvd12.ezyfoxserver.client.entity.EzyObject;

import java.util.Date;

import vn.team.freechat.constant.MessageType;

/**
 * Created by tavandung12 on 10/7/18.
 */

public class MessageReceived extends Message {

    private String from;

    public MessageReceived() {
    }

    public MessageReceived(long channelId, String from, String message) {
        super(channelId, message);
        this.from = from;
    }

    public static MessageReceived create(EzyObject data) {
        MessageReceived answer = new MessageReceived();
        answer.deserialize(data);
        return answer;
    }

    @Override
    public void deserialize(EzyObject data) {
        super.deserialize(data);
        this.from = data.get("from");
        this.sentTime = new Date();
    }

    public String getFrom() {
        return from;
    }

    @Override
    public EzyConstant getType() {
        return MessageType.RECEIVED;
    }
}

File: android/app/src/main/java/com/tvd12/freechat/data/MessageReceived.kt
package com.tvd12.freechat.data

import com.tvd12.ezyfoxserver.client.constant.EzyConstant
import com.tvd12.ezyfoxserver.client.entity.EzyObject
import com.tvd12.freechat.constant.MessageType
import java.util.*

/**
 * Created by tavandung12 on 10/7/18.
 */

class MessageReceived : Message("") {

    var from: String? = null

    companion object {

        fun create(data: EzyObject) : MessageReceived {
            val answer = MessageReceived()
            answer.deserialize(data)
            return answer
        }

    }

    override fun deserialize(data: EzyObject) {
        super.deserialize(data)
        this.from = data.get<String>("from")
        this.sentTime = Date()
    }

    override fun getType(): EzyConstant {
        return MessageType.RECEIVED
    }
}
